tsdb/db_test.go

```diff
@@ -2941,6 +2942,11 @@ func TestOpen_VariousBlockStates(t *testing.T) {
 		require.NoError(t, fileutil.Replace(dir, dir+tmpForCreationBlockDirSuffix))
 		expectedRemovedDirs[dir+tmpForCreationBlockDirSuffix] = struct{}{}
 
+		// wal Tmp blocks during creation; those should be removed on start.
+		tmpWalDir := createBlock(t, walDir, genSeries(10, 2, 30, 40))
+		require.NoError(t, fileutil.Replace(tmpWalDir, tmpWalDir+tmpLegacy))
+		expectedRemovedDirs[tmpWalDir+tmpLegacy] = struct{}{}
```

Comment:
This is testing deletion of `wal.tmp` directory, and not `wal/checkpoint.X.tmp` (which is actually what we want). You can try creating fake `.tmp` checkpoint directories instead of creating actual checkpoints and test deletion of that.